pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,401 @@
+    """
+
+    @compatibleabstractproperty
+    def namespaces(self):
+        """
+        Read-only property that returns a list with the names of the
+        namespaces defined for this repository. Note that if there were any
+        leading or trailing slash characters in namespace parameters used
+        to add the namespaces to the repository, they will be removed from
+        the namespaces returned with this property.
+
+        Returns:
+            list of :term:`string` items containing the namespace names
+        """
+        pass
+
+    @abstractmethod
+    def validate_namespace(self, namespace):
+        """
+        Validate if the namespace exists in the repository. If the
+        namespace does not exist a KeyError is raised.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          KeyError: If the namespace is not defined in the repository.
+          ValueError: if the namespace is None.
+        """
+        pass
+
+    @abstractmethod
+    def add_namespace(self, namespace):
+        """
+        Add a CIM namespace to the repository.
+
+        The namespace must not yet exist in the CIM repository.
+
+        The default connection namespace is automatically added to
+        the CIM repository upon creation of this connection.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          ValueError: If the namespace argument is None or the
+           namespace already exists.
+        """
+        pass
+
+    @abstractmethod
+    def remove_namespace(self, namespace):
+        """
+        Remove a CIM namespace from the repository.
+
+        The namespace must exist in the repository and must be empty.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          ValueError: Namespace argument is None or the repository namespace
+            is not empty
+          KeyError:  The namespace does not exist in the CIM repository.
+        """
+        pass
+
+    @abstractmethod
+    def get_class_repo(self, namespace):
```

Comment:
Since we use the term "repository" for the entity that has namespaces and all three types of CIM objects, the store for classes within such a repository should not also be called "repo". Particularly since we already introduced a term "object store" for it.

I suggest to rename this method to `get_class_store()`.

Reply:
I agree but had put it off because lots of small changes.  DONE and also for get_instance_repo and get_qualifier_repo